### PR TITLE
New is_connected algorithm

### PR DESCRIPTION
**Pull request by vitorenesduarte**

### CODE REVIEW

**Comment by vitorenesduarte** [comment]
Addresses #161

File: src/sprinter.erl [hunk 1]
```diff
@@ -405,6 +388,30 @@ populate_graph(Name, Membership, Graph) ->
...
 
+breath_first(Root, Graph, Visited0) ->
+    %% Check if every link is bidirectional
+    %% If not, stop traversal
+    In = ordsets:from_list(digraph:in_neighbours(Graph, Root)),
+    Out = ordsets:from_list(digraph:out_neighbours(Graph, Root)),
+
+    Visited1 = ordsets:union(Visited0, [Root]),
+
+    case In == Out of
```
Comment (cmeiklejohn) [inline src/sprinter.erl]: I'm not sure I understand what the check is looking for?

Comment (vitorenesduarte) [inline src/sprinter.erl]: That the links are bidirectional

**Comment by cmeiklejohn** [comment]
Can you comment the code a bit just to make it a bit clearer for the next reader as to what specifically is happening here?  Then I think we're probably good to merge it.

**Comment by cmeiklejohn** [comment]
:+1: to merge.

**Comment by cmeiklejohn** [comment]
Did this have some effect?

**Comment by vitorenesduarte** [comment]
It's way faster, that was the motivation.
The algorithm we had before would take 8 minutes to check connectedness for 512 nodes

**Pull request merged**